flatpak/com.github.Alcaro.Flips.json

```diff
@@ -0,0 +1,37 @@
+{
+    "app-id" : "com.github.Alcaro.Flips",
+    "runtime" : "org.gnome.Platform",
+    "runtime-version" : "3.34",
+    "sdk" : "org.gnome.Sdk",
+    "command" : "flips",
+    "finish-args" : [
+        /* X11 + XShm access */
+        "--share=ipc",
+        "--socket=x11",
+        /* Wayland access */
+        "--socket=wayland",
+        /* OpenGL + DRI access */
+        "--device=dri",
+        /* Needed to find the ROMs and patches */
+        "--filesystem=host:ro"
```

Comment:
Is the apply and run button tested under this sandbox?

Reply:
Running emulators will likely not work, we may make it work but it would be quite some pain for a secondary feature, especially when saving and then running the ROM works just as well. I see three options:
- dropping the relevant buttons when building in Flatpak
- making it work (could be lots of work)
- not caring and leaving the buttons there, but useless (not the best UX…)

Reply:
Considering Flips' 2x2 button GUI, dropping it would leave an ugly hole. Rearranging the GUI under Flatpak would be possible, but I'd rather not maintain two GUIs.

My favorite would be graying it out with a 'not available in Flatpak' tooltip (same for a few settings buttons), but that is not quite ideal either.

Either way, leaving broken buttons is definitely the wrong answer. Does GLib or GTK have a function to tell whether it's running in Flatpak, or would that require a custom compiler flag?

Reply:
I think there is some flag somewhere… and in the worst case we can just set it ourselfs as a build option passed by the Flatpak manifest (e.g. `-DFLATPAK` and `#ifdef FLATPAK`).